woocommerce/woocommerce
# REP: Post daily smoke test report on Slack

### Changes proposed in this Pull Request:

Closes woocommerce/woocommerce-quality#231.

This pull request:
- Creates a separate job for running the API tests in the Daily Smoke Test workflow
- Updates the daily smoke test workflow so that it would post the summary of the API test report on Slack
- Corrects some environment variables in the daily smoke test workflow
- Allows the default timeout of the API test framework to be overridden.

### How to test the changes in this Pull Request:

1. Navigate to https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml.
2. In the `Run workflow` dropdown, select this branch (`e2e/rep-daily-slack`), then click "Run workflow".
4. Navigate to the "Checks" tab of this PR. You should see the "Daily smoke test on trunk" workflow running. Wait for it to finish.
5. Whether the daily smoke test passes or fails, you should see the "Publish test reports" workflow succeeds, like so:
    <img width="937" alt="Screen Shot 2022-05-30 at 4 26 17 PM" src="https://user-images.githubusercontent.com/4509348/170950851-ee208fa6-dcf8-43cc-9d5b-9ec43fef4344.png">
    The "Publish test reports" workflow will trigger an action in the [woocommerce/woocommerce-test-reports](https://github.com/woocommerce/woocommerce-test-reports) repo to publish the daily smoke test.
6. Head over to https://github.com/woocommerce/woocommerce-test-reports/actions and verify that you see these two workflows running:
    <img width="452" alt="Screen Shot 2022-05-30 at 4 28 35 PM" src="https://user-images.githubusercontent.com/4509348/170951505-da7192f4-df38-49ce-8139-76ae5e874a63.png">
    Wait for both of them to finish.
7. Head over to https://woocommerce.github.io/woocommerce-test-reports/ and click the "API" link under "Daily smoke test"
    <img width="279" alt="Screen Shot 2022-05-30 at 4 33 52 PM" src="https://user-images.githubusercontent.com/4509348/170952173-190fa15b-e714-4c58-840a-be66a0d5da2d.png">
8. It should take you to the Allure report showing the daily API test results. Verify that the header shows `[Daily API test report]`, the last commit message in this branch, and today's date:
    <img width="622" alt="Screen Shot 2022-05-30 at 4 37 25 PM" src="https://user-images.githubusercontent.com/4509348/170953199-8b90e79a-7e2a-46b9-ac6b-e7508fcbd4be.png">
9. Verify that you receive a Slack message containing the Daily smoke test summary like this:
    <img width="465" alt="Screen Shot 2022-05-31 at 5 13 40 PM" src="https://user-images.githubusercontent.com/4509348/171138642-181e4d7e-647a-40ee-a79d-8b88d01c1470.png">
10. Check if the "View run logs" and "View full API report" buttons in the Slack message are working. They should correctly lead you to the Daily smoke test workflow run in GitHub actions, and to the Allure report for API tests respectively. "View full E2E report" will take you to a 404 page for the meantime.
11. Check if the stats mentioned in the Slack post are consistent with the Allure report in https://woocommerce.github.io/woocommerce-test-reports/daily/api/allure-report.
12. Check if the format of the Slack message is readable both in the Slack desktop and mobile apps.
13. Please review the wording on the Slack message. Let me know if there's something there that needs to be reworded, removed, or added.


### Other information:


### FOR PR REVIEWER ONLY:
